src/plugins/GUITestBase/src/GTUtilsPcrPrimerDesignForDnaAssembly.cpp

```diff
@@ -0,0 +1,367 @@
+/**
+ * UGENE - Integrated Bioinformatics Tools.
+ * Copyright (C) 2008-2021 UniPro <ugene@unipro.ru>
+ * http://ugene.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include "GTUtilsPcrPrimerDesignForDnaAssembly.h"
+
+#include <QGroupBox>
+#include <QTableWidget>
+
+#include "GTUtilsOptionPanelSequenceView.h"
+#include "GTUtilsSequenceView.h"
+#include "GTUtilsTaskTreeView.h"
+
+#include "base_dialogs/GTFileDialog.h"
+
+#include "drivers/GTMouseDriver.h"
+
+#include "primitives/GTLineEdit.h"
+#include "primitives/GTRadioButton.h"
+#include "primitives/GTSpinBox.h"
+#include "primitives/GTTableView.h"
+#include "primitives/GTWidget.h"
+
+namespace U2 {
+using namespace HI;
+
+#define GT_CLASS_NAME "GTUtilsPcrPrimerDesign"
+
+static const QString badBackboneDialogName = "UnwantedStructuresInBackboneDialog";
+
+// Parent of most widgets used in this class. Doesn't open the tab. Returns nullptr in case of an error.
+static QWidget* getTabWgt(GUITestOpStatus& os) {
+    const QWidget* inner = GTWidget::findWidget(os,
+        GTUtilsOptionPanelSequenceView::innerWidgetNames[GTUtilsOptionPanelSequenceView::PcrPrimerDesign],
+        GTUtilsSequenceView::getActiveSequenceViewWindow(os));
+    return GTWidget::findWidget(os, "runPcrPrimerDesignWidget", inner);
+}
+
+// Return the sequence generator widget. Doesn't open the tab. Returns nullptr in case of an error.
+static QWidget* getGenSeqWgt(GUITestOpStatus& os, const QWidget* tab) {
+    return GTWidget::findWidget(os, "wgtGenerateSequence", tab == nullptr ? getTabWgt(os) : tab);
+}
+
+// Return the generated sequence table. Doesn't open the tab. Returns nullptr in case of an error.
+static QTableWidget* getGenSeqTable(GUITestOpStatus& os, const QWidget* tab) {
+    return GTWidget::findExactWidget<QTableWidget*>(os, "twGeneratedSequences", getGenSeqWgt(os, tab));
+}
+
+// Return the table of result sequences. Doesn't open the tab. Returns nullptr in case of an error.
+static QTableWidget* getResultTable(GUITestOpStatus& os, const QWidget* tab) {
```

Comment:
In case of an error sets the error into "os" or some other state holder structure. Never return nullptr from getters like in all other selector methods. Use "flags" to adjust the behaviour